blog/2020-09-25-gh-release.md

```diff
@@ -0,0 +1,165 @@
+3. Building the binaries and attaching them, triggered by the `release` event
+
+This separation has several nice benefits:
+
+### Maintainability
+
+It makes the individual files smaller, making them easier to understand and maintain.
+
+### Reusability
+
+Modular workflows are easier to reuse.
+For example, we recently [added](https://github.com/comit-network/ambrosia/pull/82) automated releases to Ambrosia - a frontend for cnd that we are currently developing.
+We are not using GitFlow for Ambrosia.
+Contrary to `comit-rs`, releases in Ambrosia are triggered by creating manually creating a GitHub release.
+As such, we just needed a workflow that builds and attaches binaries to a release.
+This is exactly what the workflow (3) in `comit-rs` does and hence we were able to largely copy the design of this workflow.
+
+### Easier testing
+
+Different trigger events allows for easier testing of the individual workflows.
+Testing GitHub action workflows is usually quite the pain.
+However, the above split made it actually quite easy to trigger just the workflow I was building.
+
+### The parallel-job dilemma
+
+For our releases, we wanted to build our binaries for Linux, MacOS and Windows.
+Running [the same job on multiply platforms](https://github.com/comit-network/comit-rs/blob/acabab36058b4e3d3ae54bde10fd7ce5db66f1a5/.github/workflows/release-cnd.yml#L11-L14) is quite easy with GitHub action's [job matrix](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategy).
+By default, this executes each step in the job on each platform.
+Unfortunately, this doesn't play well with [`@actions/create-release`](https://github.com/actions/create-release) and [`@actions/upload-release-asset`](https://github.com/actions/upload-release-asset).
+
+1. The release can only be created once.
+2. The asset has to be built and uploaded for each platform.
+3. To upload the asset, we need access to the `upload_url` output of the `create-release` action.
+
+Esp. the combination if (1) and (3) means we cannot use [conditional steps](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsif) to only create the release once because we would be missing the step output for the upload.
+To compose these actions together properly, we would need to create two jobs, one for creating the release and one for building and uploading the asset.
+But to upload the asset we need the `upload_url` from the `create-release` action output.
+Step outputs are not carried over between jobs so we would to do something like writing the URL to file and use [`@actions/upload-artifact`](https://github.com/actions/upload-artifact) to move it to the next job.
+
+For me, there are certain moments in software development where - despite knowing a solution - a part of me refuses to build it because it is so absurdly complicated for what it should achieve that I keep searching for a different solution.
```

Comment:
```suggestion
For me, there are certain moments in software development where - despite knowing a solution - a part of me refuses to build it because it is so absurdly complicated so that I keep searching for a different solution.
```

generally: feels like packing German into a sentence, would simplify :)

Reply:
But that changes the meaning of my sentence fundamentally.

I am not opposed to complicated solutions in general. Only if they are too complicated for what they should achieve.